core/client/hdfs/src/main/java/alluxio/hadoop/HadoopUtils.java

```diff
@@ -44,6 +58,25 @@ public static String getPathWithoutScheme(Path path) {
     return path.toUri().getPath();
   }
 
+  /**
+   * Gets the Zookeeper addresses from the Alluxio on Zookeeper URI.
+   *
+   * @param uri the input uri to get Zookeeper addresses from
+   * @return the Zookeeper addresses
+   */
+  public static String getZookeeperAddresses(URI uri) {
+    String zookeeperAddresses;
+    try {
+      zookeeperAddresses = uri.getAuthority().substring(ZOOKEEPER_IDENTIFIER.length())
+          .replaceAll(";", ",");
+    } catch (Exception e) {
```

Comment:
could we do this in an object-oriented way by making AlluxioURI capable of parsing these new URIs? Then it can expose methods like "getAuthorityType()" and "getAuthorityData".

Reply:
AlluxioURI capable of parsing:
** normal URI alluxio://host:port/path
** zk URI alluxio://zk@host1:port1,host2:port2,host3:port3/path 
and alluxio://zk@host1:port1;host2:port2;host3:port3/path (only for Spark)
** Journal(AEE) URI alluxio://host1:port1,host2:port2,host3:port3/path

AuthorityType {
  NORMAL,
  ZOOKEEPER
  JOURNAL
}
getAuthorityData will return Zookeeper addresses when AuthorityType is ZOOKEEPER.

@aaudiber  I put isZookeeperUri(uri) and getZookeeperAddresses(uri) in the AbstractFileSystem first and then we could see whether we can make AlluxioURI capable of parsing all URI including client-side URI in a clean way.  @gpang Would it be better to make AlluxioURI capable of parsing zk URI and future jounal URI?